v2/internal/typescriptify/typescriptify.go

```diff
@@ -261,15 +275,22 @@ func (t *TypeScriptify) AddType(typeOf reflect.Type) *TypeScriptify {
 func (t *typeScriptClassBuilder) AddMapField(fieldName string, field reflect.StructField) {
 	keyType := field.Type.Key()
 	valueType := field.Type.Elem()
-	valueTypeName := valueType.Name()
+	valueTypeName := nameTypeOf(valueType)
+	if valueType.Kind() == reflect.Ptr {
```

Comment:
matches Ptr unwrapping everywhere else in this file

without this, a map value that is a pointer to a struct looses its namespace #3709